src/test/java/software/sigma/internship/service/TeacherServiceTest.java

```diff
@@ -24,34 +23,81 @@ public class TeacherServiceTest {
     @Mock
     private TeacherRepository teacherRepository;
 
-    @Spy
-    private TeacherMapper teacherMapper;
+    @Mock
+    private ModelMapper teacherMapper;
```

Comment:
I think that this approach when you Inject only tested objects and mock others is correct. So go ahead with other tests.